src/geography.hpp

```diff
@@ -75,16 +76,31 @@ class Point : public Geography {
 public:
     Point(S2GeographyPtr&& geog_ptr) : Geography(std::move(geog_ptr)){};
 
-    inline const GeographyType geog_type() const override {
+    inline GeographyType geog_type() const override {
         return GeographyType::Point;
     }
+
+    inline const S2Point& s2point() const {
+        const auto& points = static_cast<const s2geog::PointGeography&>(geog()).Points();
+        // TODO: does not work for empty point geography
+        return points[0];
+    }
 };
 
 class LineString : public Geography {
 public:
     LineString(S2GeographyPtr&& geog_ptr) : Geography(std::move(geog_ptr)){};
 
-    inline const GeographyType geog_type() const override {
+    inline GeographyType geog_type() const override {
+        return GeographyType::LineString;
+    }
+};
+
+class Polygon : public Geography {
+public:
+    Polygon(S2GeographyPtr&& geog_ptr) : Geography(std::move(geog_ptr)){};
+
+    inline GeographyType geog_type() const override {
         return GeographyType::LineString;
```

Comment:
```suggestion
        return GeographyType::Polygon;
```

Reply:
Good catch, thanks!

Reply:
Maybe add a polygon to the `test_get_type_id` test to cover this